hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconServer.java

```diff
@@ -66,6 +67,9 @@ public static void main(String[] args) {
   @Override
   public Void call() throws Exception {
     OzoneConfiguration ozoneConfiguration = createOzoneConfiguration();
+    Configuration.addDeprecation(
+        ReconServerConfigKeys.OZONE_RECON_HTTP_KEYTAB_FILE_OLD,
```

Comment:
Maybe we can add a method 'addConfigurationOverrides' in the ConfigurationProvider provider and move this logic there? When we have more deprecated keys in the future, it may be easier to maintain them there.